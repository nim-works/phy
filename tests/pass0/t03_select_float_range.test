discard """
  output: "(Done 1)"
"""
(TypeDefs
  (Int 8)
  (Float 8)
  (ProcTy (Type 0))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 2) (Locals) (Continuations
    (Continuation (Params) 0
      (Select (Type 1) (FloatVal 20.0)
        (Choice (FloatVal 15.0) (FloatVal 25.0) (Continue 1))
        (Choice (FloatVal 10.0) (Continue 2)))
    )
    (Continuation (Params) 0
      (Continue 3 (IntVal 1))
    )
    (Continuation (Params) 0
      (Continue 3 (IntVal 2))
    )
    (Continuation (Params (Type 0)))
  ))
)