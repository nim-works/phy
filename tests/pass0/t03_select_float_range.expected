.type t0 (Int)
.type t1 (Float)
.type t2 (Proc t0)
.start t2 p0
  LdImmFloat 15.0
  LdImmFloat 20.0
  LeFloat
  Branch L8 0
  LdImmFloat 20.0
  LdImmFloat 25.0
  LeFloat
  Branch L9 1
.label L8
  Jmp L11
.label L9
  LdImmInt 1
  Ret
.label L11
  LdImmInt 2
  Ret
.end
